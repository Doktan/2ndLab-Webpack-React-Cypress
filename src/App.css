.Border{
  border: 4px double black;
}
.Joke{
  color: blue;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.intro {
  background-color: grey;
}