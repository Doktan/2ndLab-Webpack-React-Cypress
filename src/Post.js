import React from "react";

function getRandomIntInclusive(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min + 1)) + min; //Максимум и минимум включаются
  }


  class Post extends React.Component {
    constructor(props){
    super(props);
    this.state = {
        num: props.num,
        userId: null,
        id: null,
        title: null,
        body: null   
      };
    }

    fetchPost() {
      fetch("https://jsonplaceholder.typicode.com/posts/" + this.props.num)
        .then(response => response.json())
        .then(result =>
          this.setState({
            userId: result.userId,
            id: result.id,
            title: result.title,
            body: result.body
          })
        )
    }

    async componentDidMount() {
      //let num = getRandomIntInclusive(1,100); // Это нужно, для выбора случайного поста в 
      /*let url = "https://jsonplaceholder.typicode.com/posts/" + this.state.num;
       await fetch(url)
        .then(res => res.json())
        .then(
          result => {
            this.setState({
              userId: result.userId,
              id: result.id,
              title: result.title,
              body: result.body
            });
          }
        )*/
        this.fetchPost()
    }


    render(){
      return(
      <div className="itro">
      <p className="Joke"><b>Случайный Пост №{this.state.num}</b></p>
      <ul>
        <li>UserId: {this.state.userId}</li>
        <li>Id: {this.state.id}</li>
        <li>Title: {this.state.title}</li>
        <li>Body: {this.state.body}</li>
      </ul>
      </div>
      )
    }
  }
  export default Post