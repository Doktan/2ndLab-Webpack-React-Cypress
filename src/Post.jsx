import React from 'react';
import PropTypes from 'prop-types';

class Post extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      num: props.num,
      userId: null,
      id: null,
      title: null,
      body: null,
    };
  }

  componentDidMount() {
    const { num } = this.state;
    this.fetchPost(num);
  }

  componentDidUpdate(prevProps) {
    const { num } = this.props;
    if (num !== prevProps.num) {
      this.fetchPost(num);
    }
  }

  fetchPost(props) {
    fetch(`https://jsonplaceholder.typicode.com/posts/${props}`)
      .then((response) => response.json())
      .then((result) => this.setState({
        num: props,
        ...result,
      }));
  }

  render() {
    const { num } = this.state;
    const { userId } = this.state;
    const { id } = this.state;
    const { title } = this.state;
    const { body } = this.state;
    return (
      <div className="itro">
        <p className="Joke">
          <b>
            Случайный Пост №
            {num}
          </b>
        </p>
        <ul>
          <li>
            UserId:
            { userId}
          </li>
          <li>
            Id:
            { id}
          </li>
          <li>
            Title:
            { title}
          </li>
          <li>
            Body:
            { body}
          </li>
        </ul>
      </div>
    );
  }
}

Post.propTypes = {
  num: PropTypes.number,
};

Post.defaultProps = {
  num: 0,
};
export default Post;
