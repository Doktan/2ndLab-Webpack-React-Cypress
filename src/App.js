import React from 'react';
import './App.css';
import Post from './Post'

function getRandomIntInclusive(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min; //Максимум и минимум включаются
}


class Comment extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      num: null,
      postId: null,
      id: null,
      name: null,
      email: null,
      body: null
    };
  }
async componentDidMount(){
    let num = getRandomIntInclusive(1,500); // Номер случайного комментария от 1 до 500
    let url = "https://jsonplaceholder.typicode.com/comments/" + num;
await fetch(url)
      .then(res => res.json())
      .then(
        result => {
          this.setState({
            num: num,
            postId: result.postId,
            id: result.id,
            name: result.name,
            email: result.email,
            body: result.body
          });
        }
      )
  }
    render(){
      return(
        <div>
          <b>Случайный комментарий №{this.state.num}</b>
          <ul>
          <li>PostId: {this.state.postId}</li>
          <li>Id: {this.state.id}</li>
          <li>Name: {this.state.name}</li>
          <li>Email: {this.state.email}</li>
          <li>Body: {this.state.body}</li>
          </ul>
        </div>
      )
    }
}

class Album extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      num: null,
      userId: null,
      id: null,
      title: null
    };
  }
async  componentDidMount(){
    let num = getRandomIntInclusive(1,100); // Номер случайного альбома 
    let url = "https://jsonplaceholder.typicode.com/albums/" + num;
 await   fetch(url)
      .then(res => res.json())
      .then(
        result => {
          this.setState({
            num: num,
            userId: result.userId,
            id: result.id,
            title: result.title
          });
        }
      )
  }
    render(){
      return(
        <div>
          <b>Случайный альбом №{this.state.num}</b>
          <ul>
          <li>UserId: {this.state.userId}</li>
          <li>Id: {this.state.id}</li>
          <li>Title: {this.state.title}</li>
          </ul>
        </div>
      )
    }
}

class Photo extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      num: null,
      albumId: null,
      id: null,
      title: null,
      url: null,
      thumbnailUrl: null
    };
  }
 async componentDidMount(){
    let num = getRandomIntInclusive(1,5000); // Случайная фотография из 5000
    let url = "https://jsonplaceholder.typicode.com/photos/" + num;
  await  fetch(url)
      .then(res => res.json())
      .then(
        result => {
        this.setState({
          num: num,
          albumId: result.albumId,
          id: result.id,
          title: result.title,
          url: result.url,
          thumbnailUrl: result.thumbnailUrl
        });
       }
      )
    }
    render(){
      return(
        <div>
          <b>Случайное Фото №{this.state.num}</b>
          <ul>
          <li>AlbumId: {this.state.albumId}</li>
          <li>Id: {this.state.id}</li>
          <li>Title: {this.state.title}</li>
          <li>Url: {this.state.url}</li>
          <li>ThumbnailUrl: {this.state.thumbnailUrl}</li>
          </ul>
        </div>
      )
    }
}

class Todo extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      num: null,
      userId: null,
      id: null,
      title: null,
      completed: null,
    };
  }
async  componentDidMount(){
    let num = getRandomIntInclusive(1,200); // Случайный список дел из 200
    let url = "https://jsonplaceholder.typicode.com/todos/" + num;
 await   fetch(url)
    .then(res => res.json())
    .then(
      result => {
        this.setState({
          num: num,
          userId: result.userId,
          id: result.id,
          title: result.title,
          completed: result.completed
        });
      }
    )
  }
  render(){
    return(
      <div>
        <b>Случайный список дел №{this.state.num}</b>
        <ul>
          <li>UserId: {this.state.userId}</li>
          <li>Id: {this.state.id}</li>
          <li>Title: {this.state.title}</li>
          <li>Completed:  {this.state.completed}</li>
        </ul>
      </div>
    )
  }
}

class User extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: null,
      name: null,
      username: null,
      email: null,
      address: {
        street: null,
        suite: null,
        city: null,
        zipcode: null,
        geo: {
          lat: null,
          lng: null
        }
      },
      phone: null,
      website: null,
      company: {
        name: null,
        catchPhrase: null,
        bs: null
      }  
    };
  }
 async componentDidMount(){
    let num = getRandomIntInclusive(1,10); // Случайный пользователь из 10
    let url = "https://jsonplaceholder.typicode.com/users/" + num;
 await   fetch(url)
    .then(res => res.json())
    .then(
      result => {
        this.setState({
          num: num,
          id: result.id,
          name: result.name,
          username: result.username,
          email: result.email,
          address: {
            street: result.address.street,
            suite: result.address.suite,
            city: result.address.city,
            zipcode: result.address.zipcode,
            geo: {
              lat: result.address.geo.lat,
              lng: result.address.geo.lng
            }
          },
          phone: result.phone,
          website: result.website,
          company: {
            name: result.company.name,
            catchPhrase: result.company.catchPhrase,
            bs: result.company.bs
          }
        });
      }
    )
  }
  render(){
    return(
      <div>
        <b>Случайный пользователь №{this.state.num}</b>
        <ul>
        <li>Id: {this.state.id}</li>
        <li>Name: {this.state.name}</li>
        <li>Username: {this.state.username}</li>
        <li>Email: {this.state.email}</li>
        <li>Address:
          <ul>
              <li>Street: {this.state.address.street}</li>
              <li>Suite: {this.state.address.suite}</li>
              <li>City: {this.state.address.city}</li>
              <li>Zipcode: {this.state.address.zipcode}</li>
              <li>Geo: 
                <ul>
                   <li>Latitude: {this.state.address.geo.lat}</li>
                   <li>Longitude: {this.state.address.geo.lng}</li>
                </ul>
                </li>
          </ul>
        </li>
        <li>Phone: {this.state.phone}</li>
        <li>Website: {this.state.website}</li>
        <li>Company:
          <ul>
              <li>Name: {this.state.company.name}</li>
              <li>CatchPhrase: {this.state.company.catchPhrase}</li>
              <li>Bs: {this.state.company.bs}</li>      
          </ul>
        </li>
        </ul>
      </div>
    )
  }
}

function Signs(props) {
  let postNum = getRandomIntInclusive(1, 10);
  console.log(postNum);
  return (
  <div>
  <h3>Пост №{props.num}</h3>
  <Post num = {props.num}/>
  <Comment/>
  <Album/>
  <Photo/>
  <Todo/>
  <User/>
  </div>
  );
}
function App() {
  return (
    <div  className="Border">
    <ul>
      <li><Signs num = "1"/></li>
      <li><Signs num = "2"/></li>
      <li><Signs num = "3"/></li>
    </ul>
    </div>
  );
}

export default App;

