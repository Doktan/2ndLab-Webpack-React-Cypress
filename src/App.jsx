import React from 'react';
import PropTypes from 'prop-types';
import './App.css';
import Post from './Post';
import Ability from './Ability';
import Comment from './Comment';
import Album from './Album';
import Photo from './Photo';
import Todo from './Todo';
import User from './User';

function Signs(props) {
  const { num } = props;
  return (
    <div>
      <h3>
        Пост №
        {num}
      </h3>
      <Post num={num} />
      <Comment />
      <Album />
      <Photo />
      <Todo />
      <User />
    </div>
  );
}

Signs.propTypes = {
  num: PropTypes.number,
};

Signs.defaultProps = {
  num: 0,
};

function App() {
  return (
    <div className="Border">
      <Ability />
      <ul>
        <li><Signs num="1" /></li>
        <li><Signs num="2" /></li>
        <li><Signs num="3" /></li>
      </ul>
    </div>
  );
}

export default App;
