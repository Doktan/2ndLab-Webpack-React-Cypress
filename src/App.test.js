import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import App from './App';
import Post from './Post';

let container = null;
beforeEach(() => {
  // подготавливаем DOM-элемент, куда будем рендерить
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // подчищаем после завершения
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders Post data", async () => {
  const fakePost = {
    userId: 1,
    id: 1,
    title: "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    body: "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
  };

  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakePost)
    })
    
  );
  await act(async () => {
    render(<Post num="1" />, container);
  });

  expect(container.querySelector("summary").textContent).toBe(fakeUser.userId);
  expect(container.querySelector("strong").textContent).toBe(fakeUser.id);
  expect(container.textContent).toContain(fakeUser.body);
  // выключаем фиктивный fetch, чтобы убедиться, что тесты полностью изолированы
  global.fetch.mockRestore();
});


/*test('class post', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Пост/i);
  expect(linkElement).toBeInTheDocument();
});*/


import renderer from 'react-test-renderer';
//import { render} from '@testing-library/react';

it('renders correctly', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


