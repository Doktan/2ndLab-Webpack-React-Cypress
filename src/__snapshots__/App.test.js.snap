// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Border"
>
  <ul>
    <li>
      <div>
        <h3>
          Пост №
          1
        </h3>
        <div
          className="itro"
        >
          <p
            className="Joke"
          >
            <b>
              Случайный Пост №
              1
            </b>
          </p>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный комментарий №
          </b>
          <ul>
            <li>
              PostId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный альбом №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайное Фото №
          </b>
          <ul>
            <li>
              AlbumId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Url: 
            </li>
            <li>
              ThumbnailUrl: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный список дел №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Completed:  
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный пользователь №
          </b>
          <ul>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Username: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Address:
              <ul>
                <li>
                  Street: 
                </li>
                <li>
                  Suite: 
                </li>
                <li>
                  City: 
                </li>
                <li>
                  Zipcode: 
                </li>
                <li>
                  Geo:
                  <ul>
                    <li>
                      Latitude: 
                    </li>
                    <li>
                      Longitude: 
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>
              Phone: 
            </li>
            <li>
              Website: 
            </li>
            <li>
              Company:
              <ul>
                <li>
                  Name: 
                </li>
                <li>
                  CatchPhrase: 
                </li>
                <li>
                  Bs: 
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <div>
        <h3>
          Пост №
          2
        </h3>
        <div
          className="itro"
        >
          <p
            className="Joke"
          >
            <b>
              Случайный Пост №
              2
            </b>
          </p>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный комментарий №
          </b>
          <ul>
            <li>
              PostId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный альбом №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайное Фото №
          </b>
          <ul>
            <li>
              AlbumId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Url: 
            </li>
            <li>
              ThumbnailUrl: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный список дел №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Completed:  
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный пользователь №
          </b>
          <ul>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Username: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Address:
              <ul>
                <li>
                  Street: 
                </li>
                <li>
                  Suite: 
                </li>
                <li>
                  City: 
                </li>
                <li>
                  Zipcode: 
                </li>
                <li>
                  Geo:
                  <ul>
                    <li>
                      Latitude: 
                    </li>
                    <li>
                      Longitude: 
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>
              Phone: 
            </li>
            <li>
              Website: 
            </li>
            <li>
              Company:
              <ul>
                <li>
                  Name: 
                </li>
                <li>
                  CatchPhrase: 
                </li>
                <li>
                  Bs: 
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <div>
        <h3>
          Пост №
          3
        </h3>
        <div
          className="itro"
        >
          <p
            className="Joke"
          >
            <b>
              Случайный Пост №
              3
            </b>
          </p>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный комментарий №
          </b>
          <ul>
            <li>
              PostId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Body: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный альбом №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайное Фото №
          </b>
          <ul>
            <li>
              AlbumId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Url: 
            </li>
            <li>
              ThumbnailUrl: 
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный список дел №
          </b>
          <ul>
            <li>
              UserId: 
            </li>
            <li>
              Id: 
            </li>
            <li>
              Title: 
            </li>
            <li>
              Completed:  
            </li>
          </ul>
        </div>
        <div>
          <b>
            Случайный пользователь №
          </b>
          <ul>
            <li>
              Id: 
            </li>
            <li>
              Name: 
            </li>
            <li>
              Username: 
            </li>
            <li>
              Email: 
            </li>
            <li>
              Address:
              <ul>
                <li>
                  Street: 
                </li>
                <li>
                  Suite: 
                </li>
                <li>
                  City: 
                </li>
                <li>
                  Zipcode: 
                </li>
                <li>
                  Geo:
                  <ul>
                    <li>
                      Latitude: 
                    </li>
                    <li>
                      Longitude: 
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>
              Phone: 
            </li>
            <li>
              Website: 
            </li>
            <li>
              Company:
              <ul>
                <li>
                  Name: 
                </li>
                <li>
                  CatchPhrase: 
                </li>
                <li>
                  Bs: 
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
